app/controllers/message_handler_controller.rb

```diff
@@ -1,5 +1,50 @@
 class MessageHandlerController < ApplicationController
+  CREATE_MESSAGE_TYPE = 'create'
+  ACTION_MESSAGE_TYPE = 'action'
+
+  before_action :initialize_creators
+
   def handle
+    message_params = params[:params]
+    case params[:type]
+      when CREATE_MESSAGE_TYPE
+        converter = PageParamConverter.new(message_params)
+        petition_params = converter.get_params_for_petition_page
+        donation_params = converter.get_params_for_donation_page
+        # We blindly create both page types, because we can use pages for
+        # both petitions and donations, and there's essentially no overhead to doing
+        # this on our end.
+        @page_creator.create_page(
+            petition_params[:name],
+            petition_params[:title],
+            petition_params[:language_uri],
+            petition_params[:url],
+            petition_params[:page_type],
+            petition_params[:page_id]
+        )
```

Comment:
this function is rather long - it would be great to DRY up these calls to create_page by making a wrapper method you can call with just `petition_params` or `donation_params`